wdl/DeNovoSVsScatter.wdl

```diff
@@ -0,0 +1,246 @@
+
+        bcftools view ~{vcf_input} | grep -v ^## | bgzip -c > ~{basename}.noheader.vcf.gz
+        python /src/denovo/denovo_svs.py \
+            --bed ~{bed_input} \
+            --ped ~{ped_input} \
+            --vcf ~{basename}.noheader.vcf.gz \
+            --disorder ~{disorder_input} \
+            --out ~{basename}.annotation.bed \
+            --out_de_novo ~{basename}.denovo.bed \
+            --raw_proband ~{raw_proband} \
+            --raw_parents ~{raw_parents} \
+            --raw_depth_proband ~{raw_depth_proband} \
+            --raw_depth_parents ~{raw_depth_parents} \
+            --config ~{python_config} \
+            --exclude_regions ~{exclude_regions} \
+            --coverage ~{batch_bincov_index} \
+            --sample_batches ~{sample_batches} \
+            --verbose True
+
+        bgzip ~{basename}.denovo.bed
+        bgzip ~{basename}.annotation.bed
+    }
+
+    runtime {
+        cpu: select_first([runtime_attr.cpu_cores, default_attr.cpu_cores])
+        memory: select_first([runtime_attr.mem_gb, default_attr.mem_gb]) + " GB"
+        disks: "local-disk " + select_first([runtime_attr.disk_gb, default_attr.disk_gb]) + " HDD"
+        bootDiskSizeGb: select_first([runtime_attr.boot_disk_gb, default_attr.boot_disk_gb])
+        preemptible: select_first([runtime_attr.preemptible_tries, default_attr.preemptible_tries])
+        maxRetries: select_first([runtime_attr.max_retries, default_attr.max_retries])
+        docker: variant_interpretation_docker
+    }
+}
+
+task VcfToBed {
+
+    input {
+        File vcf_file
+        String variant_interpretation_docker
+        RuntimeAttr? runtime_attr_override
+    }
+
+    Float input_size = size(vcf_file, "GB")
+
+    RuntimeAttr default_attr = object {
+        mem_gb: 3.75,
+        disk_gb: ceil(10 + input_size * 1.5),
+        cpu_cores: 1,
+        preemptible_tries: 2,
+        max_retries: 1,
+        boot_disk_gb: 8
+    }
+    
+    RuntimeAttr runtime_attr = select_first([runtime_attr_override, default_attr])
+
+    output {
+        File bed_output = "~{basename}.bed.gz"
+    }
+
+    String basename = basename(vcf_file, ".vcf.gz")
+    command {
+        set -exuo pipefail
+
+        svtk vcf2bed ~{vcf_file} --info ALL --include-filters ~{basename}.bed
+        bgzip ~{basename}.bed
+    }
+
+    runtime {
+        cpu: select_first([runtime_attr.cpu_cores, default_attr.cpu_cores])
+        memory: select_first([runtime_attr.mem_gb, default_attr.mem_gb]) + " GB"
+        disks: "local-disk " + select_first([runtime_attr.disk_gb, default_attr.disk_gb]) + " HDD"
+        bootDiskSizeGb: select_first([runtime_attr.boot_disk_gb, default_attr.boot_disk_gb])
+        preemptible: select_first([runtime_attr.preemptible_tries, default_attr.preemptible_tries])
+        maxRetries: select_first([runtime_attr.max_retries, default_attr.max_retries])
+        docker: variant_interpretation_docker
+    }
+}
+
+task MergeBedFiles {
```

Comment:
This task does not change the header, it reads the header from the first file and writes it to the output unchanged. The header is added by pandas `.to_csv` method in [these lines](https://github.com/VJalili/gatk-sv/blob/3b229a7303a25513bdd8427540968592272e60a3/src/denovo/denovo_svs.py#L828-L829). I suppose if we need to add the `#`, a better place would in pandas first column header.

Reply:
Right you'd have to modify the upstream step. It is better practice to have the `#` since it's more robust to identify header lines than matching on a column name.

Reply:
Sounds good, added.